src/snowcli/cli/nativeapp/init.py

```diff
@@ -238,76 +223,115 @@ def _init_with_url_and_template(
+                # Render snowflake.yml file from its jinja template
+                _render_snowflake_yml(
+                    parent_to_snowflake_yml=template_root,
+                    project_identifier=project_identifier,
+                )
+
+            # If not an official Snowflake Native App template
+            if git_url != SNOWFLAKELABS_GITHUB_URL:
+                _validate_and_update_snowflake_yml(
+                    target_directory=template_root,
+                    project_identifier=project_identifier,
+                )
+
+            project_path.parent.mkdir(parents=True, exist_ok=True)
+
+            # Move the template to the specified path
             move(
-                src=current_working_directory / temp_dir / template,
-                dst=current_working_directory / project_name,
+                src=template_root,
+                dst=project_path,
             )
 
-        path_to_project = current_working_directory / project_name
-
-        if Path.exists(path_to_project / "snowflake.yml.jinja"):
-            # Render snowflake.yml file from its jinja template
-            render_snowflake_yml(parent_to_snowflake_yml=path_to_project)
-
-        # If not an official Snowflake Native App template
-        if git_url != SNOWFLAKELABS_GITHUB_URL:
-            validate_and_update_snowflake_yml(target_directory=path_to_project)
-
+    except TemplateNotFoundError:
+        raise
     except Exception as err:
         # If there was any error, validation on Project Definition file or otherwise,
         # there should not be any Native Apps Project left after this.
-        if path_to_project:
-            rmtree(path_to_project.resolve())
+        if project_path.exists():
+            rmtree(project_path.resolve())
 
         log.error(err)
         raise InitError()
 
 
+def _is_part_of_existing_project(path: Path):
+    project_file = path / "snowflake.yml"
+    if project_file.is_file():
+        # Found a snowflake.yml in an ancestor directory, this path is part of an existing project
+        return True
+
+    parent = path.parent
+    if parent == path:
+        # reached the root without encountering an existing project
+        return False
+
+    return _is_part_of_existing_project(parent)
+
+
 def nativeapp_init(
-    name: str, git_url: Optional[str] = None, template: Optional[str] = None
+    path: str,
+    name: Optional[str] = None,
+    git_url: Optional[str] = None,
+    template: Optional[str] = None,
 ):
     """
     Initialize a Native Apps project in the user's current working directory, with or without the use of a template.
 
     Args:
+        path (str): The location of the project to be created.
         name (str): Name of the project to be created.
         git_url (str): The git URL to perform a clone from.
         template (str): A template within the git URL to use, all other directories and files outside the template will be discarded.
 
     Returns:
         None
```

Comment:
This now returns a dict, should be reflected here. Also, better to encapsulate in a dataclass so we can get type hints?